hedera-node/src/main/java/com/hedera/services/txns/contract/ContractCallTransitionLogic.java

```diff
@@ -114,11 +123,11 @@ public void doStateTransitionOperation(
 		var op = contractCallTxn.getContractCall();
 		final var target = targetOf(op);
 		final var targetId = target.toId();
-
 		// --- Load the model objects ---
 		final var sender = accountStore.loadAccount(senderId);
```

Comment:
As in `EvmTxProcessor`, we can calculate `upfrontGasCost` and check if the sender has enough balance and fail the txn if they don't have enough balance.